File: mergeSort.py

def merge(left_arry,right_arry):
    result = []
    while (len(left_arry) > 0 and len(right_arry) > 0):
        if left_arry[0] > right_arry[0]:
           result.append(right_arry.pop(0)) 
        else:
            result.append(left_arry.pop(0))
    result = result + right_arry if len(right_arry) > 0 else result + left_arry
    return result

def mergesort(arry):
    if len(arry) == 1:
        return arry
    mid = len(arry)//2
    left_arry = arry[:mid]
    right_arry = arry[mid:]
    return merge(mergesort(left_arry),mergesort(right_arry))

if __name__ =='__main__':
    sortlist=[15,669,45,51,848,6,21,48,954,877,32,54,65,3345,31,248,35,8,518]
    print(mergesort(sortlist))

File: mergeSort..js
function merge(left_arr, right_arr) {
  let result = [];
  while (left_arr.length > 0 && right_arr.length > 0) {
    if (left_arr[0] < right_arr[0]) {
      result.push(left_arr.shift());
    } else {
      result.push(right_arr.shift());
    }
  }
  result = right_arr.length > 0 ? result.concat(right_arr) : result.concat(left_arr);
  return result;
}

function mergesort(arr) {
  if (arr.length === 1) {
    return arr;
  }
  let mid = Math.floor(arr.length / 2);
  let left_arr = arr.slice(0, mid);
  let  right_arr = arr.slice(mid, arr.length);
  return merge(mergesort(left_arr), mergesort(right_arr));
}

arr = [45, 8, 9, 6, 42, 48, 0,2,-6, 95, 13, 3,68, -1, 99,-50, 5];
console.log(mergesort(arr));